src/App.js

```diff
@@ -1,16 +1,77 @@
 import React from 'react';
+import data from './data.js';
+import TodoForm from './components/TodoComponents/TodoForm';
+import Todo from './components/TodoComponents/Todo.js';
+import './components/TodoComponents/Todos.css'
+import TodoList from './components/TodoComponents/TodoList'
 
-class App extends React.Component {
+  class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor(){
+    super();
+    this.state = { //this is state
+      todoLists: data
+    }
+  }
+
+  toggleTodo = (event, todoId) => {
+    event.preventDefault();
+    this.setState({
+
+      todoLists: this.state.todoLists.map(todo => {
+        if(todo.id === todoId){
+          return {
+            ...todo,
+            completed: !todo.completed
+          }
+
+        }else{
+          return todo;
+        }
+      })
+    })
+  }
+
+  clearCompleted = (event) =>{
+    event.preventDefault()
+    this.setState({
+      todoLists: this.state.todoLists.filter( todo =>{
+        return !todo.completed;
+      })
+    })
+  }
+
+  addList = (event, todoItem) =>{
```

Comment:
Function is fine, but be mindful of what you're calling your functions.  addItem might be a better name vs addList